lib/rex/proto/http/client.rb

```diff
@@ -234,10 +237,11 @@ def _send_recv(req, t = -1, persist = false)
     elsif req.respond_to?(:opts) && req.opts['krb_transform_request'] && self.krb_encryptor
       req = req.opts['krb_transform_request'].call(self.krb_encryptor, req)
     end
-
+    subscriber.on_request(req)
```

Comment:
We should move this above line 235:

```ruby
    subscriber.on_request(req)
    if req.respond_to?(:opts) && req.opts['ntlm_transform_request'] && self.ntlm_client
      req = req.opts['ntlm_transform_request'].call(self.ntlm_client, req)
    elsif req.respond_to?(:opts) && req.opts['krb_transform_request'] && self.krb_encryptor
      req = req.opts['krb_transform_request'].call(self.krb_encryptor, req)
    end
```

So when we httptrace the winrm login module, the raw soap request is logged - instead of the encrypted request:


### Before

The encrypted binary request is logged

```
####################
# Request:
####################
POST /wsman HTTP/1.1
Host: 192.168.123.13:5985
User-Agent: Microsoft WinRM Client
Content-Type: multipart/encrypted;protocol="application/HTTP-SPNEGO-session-encrypted";boundary="Encrypted Boundary"
Content-Length: 2102

--Encrypted Boundary
Content-Type: application/HTTP-SPNEGO-session-encrypted
OriginalContent: type=application/soap+xml;charset=UTF-8;Length=1847
--Encrypted Boundary
Content-Type: application/octet-stream
?d????W???x
           9Gh;'0??xZ{A
                       *??f?vG???!?YM?????d??4W~????
>8?D޷????s??l%J'?>gG???5)U@?հ?L^?[R\??
??{}H?|?????pw????߀????U????F9'??PP??

```

### After

The raw request before encryption is logged:

```
####################
# Request:
####################
POST /wsman HTTP/1.1
Host: 192.168.123.13:5985
User-Agent: Microsoft WinRM Client
Content-Type: application/soap+xml;charset=UTF-8
Content-Length: 1895

<?xml version="1.0" encoding="UTF-8"?><env:Envelope xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:env="http://www.w3.org/2003/05/soap-envelope" xmlns:a="http://schemas.xmlsoap.org/ws/2004/08/addressing" xmlns:b="http://schemas.dmtf.org/wbem/wsman/1/cimbinding.xsd" xmlns:n="http://schemas.xmlsoap.org/ws/2004/09/enumeration" xmlns:x="http://schemas.xmlsoap.org/ws/2004/09/transfer" xmlns:w="http://schemas.dmtf.org/wbem/wsman/1/wsman.xsd" xmlns:p="http://schemas.microsoft.com/wbem/wsman/1/wsman.xsd" xmlns:rsp="http://schemas.microsoft.com/wbem/wsman/1/windows/shell" xmlns:cfg="http://schemas.microsoft.com/wbem/wsman/1/config"><env:Header><a:To>http://192.168.123.13:5985/wsman</a:To><a:ReplyTo><a:Address mustUnderstand="true">http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous</a:Address></a:ReplyTo><w:MaxEnvelopeSize mustUnderstand="true">153600</w:MaxEnvelopeSize><a:MessageID>uuid:E903F107-90C7-4CF9-89A2-031EE5E1B453</a:MessageID><p:SessionId mustUnderstand="false">uuid:99D1558F-FA62-4911-9AB6-2FB96034BE23</p:SessionId><w:Locale xml:lang="en-US" mustUnderstand="false"/><p:DataLocale xml:lang="en-US" mustUnderstand="false"/><w:OperationTimeout>PT1S</w:OperationTimeout><w:ResourceURI mustUnderstand="true">http://schemas.microsoft.com/wbem/wsman/1/windows/shell/cmd</w:ResourceURI><a:Action mustUnderstand="true">http://schemas.microsoft.com/wbem/wsman/1/windows/shell/Command</a:Action><w:OptionSet><w:Option Name="WINRS_CONSOLEMODE_STDIN">TRUE</w:Option><w:Option Name="WINRS_SKIP_CMD_SHELL">FALSE</w:Option></w:OptionSet><w:SelectorSet><w:Selector Name="ShellId">AE21BDC9-2A9F-4009-8438-1F2DC6058E09</w:Selector></w:SelectorSet></env:Header><env:Body><rsp:CommandLine CommandId="D139BB66-A6C8-4CA9-A79A-55C1EC4A1293"><rsp:Command>&quot;cmd.exe&quot;</rsp:Command></rsp:CommandLine></env:Body></env:Envelope>
```